README.md

```diff
@@ -24,3 +33,96 @@ For the documentation of the module in `README.md`, the following guidelines app
+**Post-install module config** is used to allow local-users to see & change the behavior of the module on booted system, in order to improve local-user experience.
+
+Example of the module which satisfies the requirements & implements this functionality: `default-flatpaks`
+
+#### Config Requirements
+**Requirements** for post-install module configs exist, as not all modules need this functionality.  
+Following conditions for approved post-install module config implementation are:
+
+- **module performs it's functions on booted system**  
+ Modules which are fully utilized in build-time don't need configuration options, as those are already located in `recipe.yml`.
+- **post-install module config can be implemented without affecting reliability of the system**  
+ Module-maintainer needs to carefully select which type of module to implement based on condition above. If a module compromises system reliability when used on booted system, making the module build-time based should be considered. Examples of this are `rpm-ostree` & `akmods` modules, which are better utilized as build-time modules.
+- **module can have additional useful options for configuring**  
+Which can improve local-user experience.
+- **module can strongly collide with local-user's usage pattern with it's default behavior**  
+Example: `default-flatpaks` module can remove a flatpak app, which local-user used daily.
+
+#### Config Format
+
+In order to keep config files easy to read & reliable to parse, standardized `.yml` markup format is used.  
+[`yq`](https://github.com/mikefarah/yq) tool is used to process `.yml` configs in order to reach the desired goal.
+
+#### Config Directory Structure
+
+**System config:**  
+`/usr/share/bluebuild/module-name/config.yml`
+
+**System config** is a post-install module config which is derived from `recipe.yml` module entry. It is placed in this read-only directory location in order to avoid local-users writing to it. System config is purely intended to inform local-users about which modifications are done in `recipe.yml`, so they can potentially proceed with modifications on their own.
+
+**Local-user config:**  
+`/usr/etc/bluebuild/module-name/config.yml`
+
+**Local-user config** is a post-install module config which is derived from local-user config template. It is placed in `/usr/etc`, which is then automatically copied to `/etc`, which is writable to local-users. `/usr/etc` local-user config can be used to reset module config that is done in `/etc`.
+
+#### Config Example
+
+System config (`/usr/share/bluebuild/default-flatpaks/config.yml`):
+
+```yaml
+vendor: BlueBuild
+module: default-flatpaks
+description: System config file for `default-flatpaks` BlueBuild module, which is used to install + remove flatpak apps or modify flatpak repos.
+instructions: Read this system config in order to know what is currently configured by the system & what to potentially modify in local-user config (/etc/bluebuild/default-flatpaks/config.yml).
```

Comment:
What's the point of this section? Just to inform the reader? Wouldn't comments in the YAML file be a more standard way to do that?

Reply:
Yeah, with this it seems that these properties would be used for something. The `vendor` and `module` keys are already in the file path too, but idk...

Reply:
I just added # as comments for those entries & "# Information about the config file" above.

Tell me if you have a better suggestion.

Reply:
Looks great!